### ISSUE TITLE
RPM files generate errors

### ISSUE THREAD

**Issue by pointo1d** [anchore/syft#2221]
**What happened**:
Having downloaded & installed (`sudo yum install ...`) the syft RPM (`syft_0.93.0_linux_amd64.rpm`), I ran it on a mounted RHEL 8u6 distro ISO image (from which the hosting VM was itself successfully installed) and observed the following...
```
syft dir:/mnt/cdrom
✔ Indexed file system
⠴ Cataloging packages             [0 packages]  
[0030]  WARN cataloger failed cataloger=rpm-file-cataloger error=RPM file found but unable to read: /AppStream/Packages/at-spi2-core-devel-2.28.0-1.el8.x86_64.rpm (file is not an RPM) location=/AppStream/Packages/at-spi2-core-devel-2.28.0-1.el8.x86_64.rpm
[0030]  WARN cataloger failed cataloger=rpm-file-cataloger error=RPM file found but unable to read: /AppStream/Packages/atinject-1-28.20100611svn86.module+el8+2452+b359bfcd.noarch.rpm (file is not an RPM) location=/AppStream/Packages/atinject-1-28.20100611svn
[0030]  WARN cataloger failed cataloger=rpm-file-cataloger error=RPM file found but unable to read: /AppStream/Packages/atinject-1-31.20100611svn86.module+el8.6.0+13337+afcb49ec.noarch.rpm (file is not an RPM) location=/AppStream/Packages/atinject-1-31.201006
.
.
.
```

Upon observing the above, I aborted the attempt before completion.

**What you expected to happen**:
The discovered licenses to be catalogued.

**Steps to reproduce the issue**:
syft dir:/mnt/cdrom

**Anything else we need to know?**:
RHEL 8u6 VM running in VMWare Workstation Pro 16.1.1 build-17801498

**Environment**:
- Output of `syft version`:
`syft 0.93.0`

- OS (e.g: `cat /etc/os-release` or similar):
```
NAME="Red Hat Enterprise Linux"
VERSION="8.6 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.6"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.6 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.6
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.6"
```

**Comment by pointo1d** [comment]
Seems that the mount was stale - `sudo umount /mnt/cdrom ; sudo mount ... /mnt/cdrom` seems to have fixed the problem however, I expected a list of products + licences, instead of which the report was merely a list of RPMs (that I could have used a simple `ls`(1) for)?!

**Comment by tgerla** [comment]
Hi @pointo1d, are you looking at the default (table format) output from Syft? That won't show license information. You can generate the SBOM in syft-json format which includes a lot more information using the ```--output syft-json``` flag. You can also design your own template if you have specific requirements for the output format. (https://github.com/anchore/syft#using-templates)

Hope this helps--let us know if you need anything else.

**Comment by pointo1d** [comment]
TFT Tim,

It did help, the analysis got further but failed to discover embedded licences within nested containers i.e. container images inside tgz inside RPM (yeah, I know, but what can we do?? ;-)) - or is that something (currently) outside the scope of syft. FWIW, we're having to run post-build ISO image scans .

**Comment by tgerla** [comment]
Hey @pointo1d, yeah, we don't currently recursively scan into nested containers. However, if you are able to scan and save an SBOM file alongside those containers, Syft can find them when you're scanning the outer container. For instance:

```
outer container:
  - image-1.sbom
  - image-1.tgz
  - image-2.sbom
  - image-2.tgz
```

Scanning the outer container will then result in a combined SBOM with all of the contents from the outer container and the two inner images. Does this help?

**Comment by pointo1d** [comment]
It does thank you ... immeasurably

**Comment by spiffcs** [comment]
For the related issue please see: https://github.com/anchore/syft/issues/1379

@pointo1d I think your current path has been unblocked and the desired feature can be found ready to be implemented in #1379.

Closing, but if you have more questions happy to reopen.
